pkgs/tools/package-management/packagekit/default.nix

```diff
@@ -1,52 +1,82 @@
-{ stdenv, fetchFromGitHub, lib
-, intltool, glib, pkg-config, polkit, python3, sqlite
-, gobject-introspection, vala, gtk-doc, autoreconfHook, autoconf-archive
-, nix, enableNixBackend ? false, boost
+{ stdenv
+, fetchFromGitHub
+, lib
+, intltool
+, glib
+, pkg-config
+, polkit
+, python3
+, sqlite
+, gobject-introspection
+, vala
+, gtk-doc
+, nix
+, boost
+, meson
+, ninja
+, nlohmann_json
+, libxslt
+, libxml2
+, gst_all_1
+, gtk3
 , enableCommandNotFound ? false
-, enableBashCompletion ? false, bash-completion ? null
-, enableSystemd ? stdenv.isLinux, systemd }:
+, enableBashCompletion ? false
+, bash-completion ? null
+, enableSystemd ? stdenv.isLinux
+, systemd
+}:
 
 stdenv.mkDerivation rec {
   pname = "packagekit";
-  version = "1.1.13";
+  version = "1.2.5pre";
 
   outputs = [ "out" "dev" ];
 
   src = fetchFromGitHub {
     owner = "hughsie";
     repo = "PackageKit";
-    rev = "PACKAGEKIT_${lib.replaceStrings ["."] ["_"] version}";
-    sha256 = "0xmgac27p5z8wr56yw3cqhywnlvaf8kvyv1g0nzxnq167xj5vxam";
+    rev = "e8140563ec7f78dd723068396dcad40e47580379";
+    sha256 = "0mz54km3fy1rlbldz2nj778022i84gvvy6hyk2jym815qlvr3mkq";
   };
 
-  buildInputs = [ glib polkit python3 gobject-introspection ]
-                  ++ lib.optional enableSystemd systemd
-                  ++ lib.optional enableBashCompletion bash-completion;
-  propagatedBuildInputs =
-    [ sqlite boost ]
-    ++ lib.optional enableNixBackend nix;
-  nativeBuildInputs = [ vala intltool pkg-config autoreconfHook autoconf-archive gtk-doc ];
-
-  preAutoreconf = ''
-    gtkdocize
-    intltoolize
-  '';
+  buildInputs = [
+    glib
+    polkit
+    python3
+    gobject-introspection
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gtk3
+    nlohmann_json
+  ] ++ lib.optional enableSystemd systemd
+  ++ lib.optional enableBashCompletion bash-completion;
+  propagatedBuildInputs = [
+    sqlite
+    boost
+    nix
+  ];
+  nativeBuildInputs = [
+    vala
+    intltool
+    pkg-config
+    gtk-doc
+    meson
+    libxslt
+    libxml2
+    ninja
+  ];
 
-  configureFlags = [
-    (if enableSystemd then "--enable-systemd" else "--disable-systemd")
-    "--disable-dummy"
-    "--disable-cron"
-    "--enable-introspection"
-    "--disable-offline-update"
-    "--localstatedir=/var"
-    "--sysconfdir=/etc"
```

Comment:
I am afraid PackageKit will use configs from `${out}/etc` without this. We might either patch the runtime paths ([example](https://github.com/NixOS/nixpkgs/blob/dc7d431cd8f4dbd5c35698c68a0e9215176ac669/pkgs/os-specific/linux/firmware/fwupd/default.nix#L128-L133)), or hack around this using `DESTDIR` ([example](https://github.com/NixOS/nixpkgs/blob/dc7d431cd8f4dbd5c35698c68a0e9215176ac669/pkgs/desktops/gnome/core/gdm/default.nix#L139-L161)).

Reply:
should it use configs from /etc? Where should the state be stored?

I can't figure out how it does installs, does it manage the user profile via nix-env?